service-workers/service-worker/fetch-event-request-abort-signal.https.html

```diff
@@ -0,0 +1,165 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Fetch event request abort signal</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<script>
+const SW_URL = 'resources/abort-monitor-worker.js';
+const SCOPE = 'resources/blank.html';
+
+function raf() {
+  return new Promise(r => requestAnimationFrame(r));
+}
+
+async function nextFrame() {
+  await raf();
+  await raf();
+}
+
+function broadcastResponse(channel) {
+  const bc = new BroadcastChannel(channel);
+  const p = new Promise(resolve => {
+    bc.onmessage = event => {
+      bc.close();
+      resolve(event.data);
+    };
+  });
+
+  p.broadcastChannel = bc;
+
+  return p;
+}
+
+function reset() {
+  for (const iframe of [...document.querySelectorAll('.test-iframe')]) {
+    iframe.remove();
+  }
+  return navigator.serviceWorker.getRegistrations().then(registrations => {
+    return Promise.all(registrations.map(r => r.unregister()));
+  });
+}
+
+promise_test(async t => {
+  await reset();
```

Comment:
Shouldn't this be a test_cleanup step of sorts?

Reply:
Unfortunately it doesn't support async cleanup steps

Reply:
Aah okay. A single comment to that effect would be good.

Reply:
And maybe an issue to add promise_cleanup?